rcl/src/rcl/wait.c

```diff
@@ -79,6 +80,9 @@ __wait_set_clean_up(rcl_wait_set_t * wait_set, rcl_allocator_t allocator)
     assert(ret == RCL_RET_OK);  // Defensive, shouldn't fail with size 0.
   }
   if (wait_set->impl) {
+    if (wait_set->impl->rmw_waitset) {
+      assert(rmw_destroy_waitset(wait_set->impl->rmw_waitset) == RMW_RET_OK);
```

Comment:
ok, I've moved `rmw_destroy_waitset` out of the assert statement so that it better follows the pattern established in the above code. I'll leave changing the assert statement until after we've resolved the question about the custom assert.

Consider that in if the assert is truly defensive, it may be just fine to only check it in Debug mode (if the system is well tested) and ignore the assert in Release mode for performance purposes.